### ISSUE TITLE
Last user should not be root does not consider parent image

### ISSUE THREAD

**Issue by nmoreau-ocl** [hadolint/hadolint#648]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
DL3002 (aka last user should not be root) should fail why the parent image is the running as the root user and the child image has not used the USER directive to change it

### Actual behavior
DL3002 passes scanning child images that are configured to run as root

### Steps to reproduce the behavior

**Output of `hadolint --version` 

```bash
Haskell Dockerfile Linter 2.4.0-no-git
```

**Dockerfile** (if relevant)

```Dockerfile
FROM container-registry.oracle.com/os/oraclelinux:8
```

**Additional environment details (OS, stack version, etc.)**

**Comment by sxmartin** [comment]
Hello.. We are also running into this issue, where hadolint is not failing if the docker image is configured to run as root.  Appreciate update on this.. -thanks

**Comment by pascal-hofmann** [comment]
See also #1089.
